s2spy/preprocess.py

```diff
@@ -8,7 +8,12 @@
 import xarray as xr
 
 
-TIMESCALE_TYPE = Literal["monthly", "weekly", "daily"]
+try:
+    from typing import TypeAlias
+except ImportError:
+    from typing_extensions import TypeAlias
+
+Timescale: TypeAlias = Literal["monthly", "weekly", "daily"]
```

Comment:
This sadly doesn't show up nicely in the api. I think that would require autodoc (https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_type_aliases). Autodoc then also requires typing out which functions/classes you want documented.

Perhaps just replace-all `Timescale` -> `Literal["monthly", "weekly", "daily"]`. A lot less work than changing the docs.

The autoapi lack of type alias support is discussed here: https://github.com/readthedocs/sphinx-autoapi/issues/224. One of the issues is that autoapi just reads the source code files, while autodoc actually imports modules. So I don't see this fixed anytime soon.

Reply:
> This sadly doesn't show up nicely in the api. I think that would require autodoc (https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_type_aliases). Autodoc then also requires typing out which functions/classes you want documented.
> 
> Perhaps just replace-all `Timescale` -> `Literal["monthly", "weekly", "daily"]`. A lot less work than changing the docs.
> 

I guess there is no better way. I will replace `Timescale` with things all spelling out.

> The autoapi lack of type alias support is discussed here: [readthedocs/sphinx-autoapi#224](https://github.com/readthedocs/sphinx-autoapi/issues/224). One of the issues is that autoapi just reads the source code files, while autodoc actually imports modules. So I don't see this fixed anytime soon.

😒 OK....too bad...